src/MutableMap/MutableMap.dfy

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+*  Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT
+*******************************************************************************/
+
+// RUN: %dafny /compile:0 "%s"
+
+include "MutableMapTrait.dfy"
+
+module {:extern "DafnyLibraries"} {:options "-functionSyntax:4"} MutableMap refines MutableMapTrait {
```

Comment:
Having a real mismatch between the Dafny and extern names for a module seems like trouble, can we nest it in Dafny as `DafnyLibraries.MutableMap` instead?

Reply:
On the Java side, we have a class `MutableMap` within a package `DafnyLibraries`. I am now following the same pattern and naming in Dafny. I don't think having two nested modules within Dafny works, if the Java side only has one module.

Reply:
Actually, importing in Dafny via `import opened DafnyLibraries` seems odd. So either: 
i) we find a way to use nested modules (which I don't think works with `:extern`, since we only have one on the Java side)
ii) we stick to the original solution, which used the name `MutableMap` in Dafny, and linked it to `DafnyLibraries` in Java
iii) we change the package name `DafnyLibraries` to `MutableMap` on the Java side